src/conf.c

```diff
@@ -804,6 +826,16 @@ bool main_config_write(meshlink_handle_t *mesh, const char *conf_subdir, const c
 		return false;
 	}
 
+	// Sync the renamed main conf file's directory where the file system looks to find your file
+	// and to avoid system failure (such as power loss, ENOSPC or an I/O error)
+	snprintf(path, sizeof(path), "%s" SLASH "%s", mesh->confbase, conf_subdir);
+
+	if(!sync_path(path)) {
```

Comment:
All functions that call `main_config_write()` will already call `config_sync()` themselves, which in turn will sync this path.